napari/_qt/qt_viewer.py

```diff
@@ -452,6 +527,19 @@ def console(self):
                     self.console.push(
                         {'napari': napari, 'action_manager': action_manager}
                     )
+                    for i in self.console_backlog:
+                        if isinstance(i, dict):
+                            # recover weak refs
+                            self.console.push(
+                                {
+                                    k: self._unwrap_if_weakref(v)
+                                    for k, v in i.items()
+                                    if self._unwrap_if_weakref(v) is not None
+                                }
+                            )
+                        else:
+                            self.console.push(i)
```

Comment:
@davidpross I think this line isn't covered by tests? Ditto for line 508. tbh I don't understand the usage that would get us there. Could you elaborate on that and maybe add a test? I think this is very close to merging otherwise! 🙏

Reply:
Good catch on the test coverage. It turns out that passing a variable name as a string or list of variable names was not working in the context of a console that is not yet instantiated. I added code from [here](https://github.com/ipython/ipython/blob/b3cd3b056a90b5d553746e96628aeb6dbb7b37dd/IPython/core/interactiveshell.py#L1517) to handle these cases and updated the unit tests accordingly. Also the console_backlog will now always be a list of dicts.